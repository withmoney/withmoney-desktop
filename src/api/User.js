import axios from 'axios';

export const login = data => axios.post('http://localhost:3000/api/v1/login', data);

export const signup = data => axios.post('http://localhost:3000/api/v1/signup', data);
