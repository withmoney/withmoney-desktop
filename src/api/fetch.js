import axios from 'axios';
import { catchInvalidToken } from 'app/utils/parse';

const newAxios = axios.create({
  baseURL: `${process.env.MYMONEY_API}`,
  headers: {
    Authorization: {
      toString() {
        return `Bearer ${localStorage.getItem('token')}`;
      },
    },
  },
});

const fetch = method => (...args) => catchInvalidToken(newAxios[method](...args));

export default {
  get: fetch('get'),
  post: fetch('post'),
  put: fetch('put'),
  delete: fetch('delete'),
};
