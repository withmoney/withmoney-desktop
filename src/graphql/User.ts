import { gql } from '@apollo/client';

export const UPDATE_USER = gql`
  mutation updateUser($firstName: String!, $lastName: String!, $email: String!, $language: Locale) {
    updateOneUser(
      data: { firstName: $firstName, lastName: $lastName, email: $email, language: $language }
    ) {
      id
      firstName
      lastName
      email
      language
    }
  }
`;
