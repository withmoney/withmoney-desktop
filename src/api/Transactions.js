import fetch from 'api/fetch';
import { mountQuery } from 'app/utils/parse';

export const list = (query = {}) => fetch.get(mountQuery('transactions', query));

export const get = id => fetch.get(`transactions/${id}`);

export const put = (id, data) => fetch.put(`transactions/${id}`, data);
