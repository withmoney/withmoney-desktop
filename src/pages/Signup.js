import React, { Fragment } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { Helmet } from 'react-helmet';
import * as UserApi from 'api/User';
import * as UserAction from 'store/user';
import BoxForm from 'components/BoxForm';
import FieldInput from 'components/FieldInput';
import FieldButton from 'components/FieldButton';

class Signup extends React.Component {
  constructor(props) {
    super(props);

    this.handleChange = this.handleChange.bind(this);
    this.onSave = this.onSave.bind(this);

    this.state = {
      name: '',
      email: '',
      password: '',
      passwordConfirm: '',
    };
  }

  async onSave(event) {
    event.preventDefault();
    const { password, passwordConfirm } = this.state;

    if (password !== passwordConfirm) {
      throw new Error('please confirm your password');
    }

    try {
      await UserApi.signup(this.state);

      console.log(
        'Your signup with success, please wait til the administratos enabled you account.',
      );
    } catch (e) {
      console.log("Error, It's not possible create your account at this moment.");
    }
  }

  handleChange({ target }) {
    this.setState({
      [target.id]: target.value,
    });
  }

  renderFields() {
    return (
      <Fragment>
        <FieldInput id="name" onChange={this.handleChange} placeholder="name" />
        <FieldInput id="email" onChange={this.handleChange} placeholder="email" />
        <FieldInput
          id="password"
          type="password"
          onChange={this.handleChange}
          placeholder="password"
        />
        <FieldInput
          id="passwordConfirm"
          type="password"
          onChange={this.handleChange}
          placeholder="confirm your password"
        />
        <FieldButton type="submit">Sign Up</FieldButton>
      </Fragment>
    );
  }

  render() {
    const footer = (
      <Fragment>
        <span>Do Already have an account?</span>
        <Link to="/login">Log In</Link>
      </Fragment>
    );

    return (
      <Fragment>
        <Helmet>
          <title>Signup</title>
          <body className="page-signup" />
        </Helmet>
        <BoxForm
          title="withmoney"
          subtitle="Sign Up"
          onSubmit={this.onSave}
          fields={this.renderFields()}
          footer={footer}
        />
      </Fragment>
    );
  }
}

const mapStateToProps = ({ user }) => ({ user });

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(UserAction, dispatch),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Signup);
