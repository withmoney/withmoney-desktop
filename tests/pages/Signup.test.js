import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { mount } from 'enzyme';
import Signup from 'pages/Signup';
import * as UserApi from 'api/User';

const mockStore = configureStore([thunk]);

jest.mock('api/User', () => ({
  signup: jest.fn().mockResolvedValue({
    data: {
      success: true,
    },
  }),
}));

describe('Signup', () => {
  let Component;

  beforeAll(() => {
    Component = global.withReduxAndRouter(mockStore({}), Signup);
  });

  it('should render a component correctly', () => {
    const wrapper = renderer.create(Component());

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should call onSabe action when signup form is submitted', () => {
    const wrapper = mount(Component());

    wrapper
      .find('input[name="name"]')
      .simulate('change', { target: { name: 'name', value: 'David Costa' } });
    wrapper
      .find('input[name="email"]')
      .simulate('change', { target: { name: 'email', value: 'davidcostadev@gmail.com' } });
    wrapper
      .find('input[name="password"]')
      .simulate('change', { target: { name: 'password', value: '123456' } });
    wrapper
      .find('input[name="passwordConfirm"]')
      .simulate('change', { target: { name: 'passwordConfirm', value: '123456' } });

    wrapper.find('form').simulate('submit');

    expect(UserApi.signup).toBeCalledWith({
      name: 'David Costa',
      email: 'davidcostadev@gmail.com',
      password: '123456',
      passwordConfirm: '123456',
    });
  });
});
