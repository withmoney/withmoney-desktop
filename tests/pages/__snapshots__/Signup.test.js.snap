// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup should render a component correctly 1`] = `
<div
  className="box-form"
>
  <h1
    className="box-form__title"
  >
    withmoney
  </h1>
  <form
    className="box-form__form"
    onSubmit={[Function]}
  >
    <h3
      className="box-form__subtitle"
    >
      Sign Up
    </h3>
    <div
      className="field"
    >
      <input
        className="field__input"
        id="name"
        name="name"
        onChange={[Function]}
        placeholder="name"
        type="text"
      />
    </div>
    <div
      className="field"
    >
      <input
        className="field__input"
        id="email"
        name="email"
        onChange={[Function]}
        placeholder="email"
        type="text"
      />
    </div>
    <div
      className="field"
    >
      <input
        className="field__input"
        id="password"
        name="password"
        onChange={[Function]}
        placeholder="password"
        type="password"
      />
    </div>
    <div
      className="field"
    >
      <input
        className="field__input"
        id="passwordConfirm"
        name="passwordConfirm"
        onChange={[Function]}
        placeholder="confirm your password"
        type="password"
      />
    </div>
    <div
      className="field"
    >
      <button
        className="button"
        type="submit"
      >
        Sign Up
      </button>
    </div>
    <div
      className="box-form__footer"
    >
      <span>
        Do Already have an account?
      </span>
      <a
        href="/login"
        onClick={[Function]}
      >
        Log In
      </a>
    </div>
  </form>
</div>
`;
